packages/env/lib/config/test/parse-config.js

```diff
@@ -0,0 +1,336 @@
+'use strict';
+/* eslint-disable jest/no-conditional-expect */
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+/**
+ * Internal dependencies
+ */
+const { parseConfig } = require( '../parse-config' );
+const readRawConfigFile = require( '../read-raw-config-file' );
+const { getLatestWordPressVersion } = require( '../../wordpress' );
+const { ValidationError } = require( '../validate-config' );
+const detectDirectoryType = require( '../detect-directory-type' );
+
+jest.mock( 'got', () => jest.fn() );
+jest.mock( '../read-raw-config-file', () => jest.fn() );
+jest.mock( '../detect-directory-type', () => jest.fn() );
+jest.mock( '../../wordpress', () => ( {
+	getLatestWordPressVersion: jest.fn(),
+} ) );
+
+/**
+ * Since our configurations are merged, we will want to refer to the parsed default config frequently.
+ */
+const DEFAULT_CONFIG = {
+	port: 8888,
+	testsPort: 8889,
+	phpVersion: null,
+	coreSource: {
+		type: 'git',
+		url: 'https://github.com/WordPress/WordPress.git',
+		ref: '100.0.0',
+		path: '/cache/WordPress',
+		clonePath: '/cache/WordPress',
+		basename: 'WordPress',
+		testsPath: '/cache/tests-WordPress',
+	},
+	pluginSources: [],
+	themeSources: [],
+	config: {
+		WP_DEBUG: true,
+		SCRIPT_DEBUG: true,
+		WP_ENVIRONMENT_TYPE: 'local',
+		WP_PHP_BINARY: 'php',
+		WP_TESTS_EMAIL: 'admin@example.org',
+		WP_TESTS_TITLE: 'Test Blog',
+		WP_TESTS_DOMAIN: 'localhost',
+		WP_SITEURL: 'http://localhost',
+		WP_HOME: 'http://localhost',
+	},
+	mappings: {},
+	env: {
+		development: {},
+		tests: {
+			config: {
+				WP_DEBUG: false,
+				SCRIPT_DEBUG: false,
+			},
+		},
+	},
+};
+
+describe( 'parseConfig', () => {
+	beforeEach( () => {
+		readRawConfigFile.mockResolvedValue( null );
+		detectDirectoryType.mockResolvedValue( null );
+		getLatestWordPressVersion.mockResolvedValue( '100.0.0' );
+	} );
+
+	afterEach( () => {
+		jest.clearAllMocks();
+		delete process.env.WP_ENV_PORT;
+		delete process.env.WP_ENV_TESTS_PORT;
+		delete process.env.WP_ENV_CORE;
+		delete process.env.WP_ENV_PHP_VERSION;
+	} );
+
+	it( 'should return default config', async () => {
+		const parsed = await parseConfig( './', '/cache' );
+
+		expect( parsed ).toEqual( DEFAULT_CONFIG );
+	} );
+
+	it( 'should infer a core mounting default when ran from a WordPress directory', async () => {
+		detectDirectoryType.mockResolvedValue( 'core' );
+
+		const parsed = await parseConfig( './', '/cache' );
+
+		expect( parsed ).toEqual( {
+			...DEFAULT_CONFIG,
+			coreSource: {
+				basename: 'gutenberg',
```

Comment:
@ObliviousHarmony @noahtallen One side-effect of hard-coding the basename (and also testPath) like this is is that if someone checks out the gutenberg repo into a differently named folder, these tests will fail.

This is currently causing a few issues for CI jobs that run on security patches, as those are created on a differently named private repo.

I was wondering what your thoughts on a good fix would be?

A naive option could be `expect.any( String )`, but I think maybe this should all be mocked rather than being based off of the real file system.

Reply:
That's an interesting case @talldan. It seems like the best source would be to just generally avoid any relative paths and use fake absolute paths in our tests. [I've gone ahead and created an issue for this](https://github.com/WordPress/gutenberg/issues/50800).

Reply:
Thanks!